internal/tracker/tracker.go

```diff
@@ -314,20 +302,23 @@ func filterRecords(records []taskRecord, prune bool) []taskRecord {
 	return relevant
 }
 
-// setExpiration sets the expires_on attribute on records. The expires_on value for record N is the
-// effective_on value of the n-1 record. The first record on the list does not contain an expires_on
-// value since there is no newer record that invalidates it in the future.
-// TODO: Probably need to compute the expires_on value without requiring the "effective_on" value so
-// we don't have to always require both values. But this may be required during some transition
-// period.
-func (t *Tracker) setExpiration() {
+// setExpiration sets the expires_on attribute on records using a duration-based approach.
+// Each record expires after a configurable duration based on inEffectDays from when it's added.
+// The most recent record (index 0) gets no expiration date, making it the current active record.
+func (t *Tracker) setExpiration(inEffectDays int) {
```

Comment:
As explained above, the new way of adding the expires_on field is to set the same value, taken from the `--in-effect-days` flag, to all the older records except the latest one.

Reply:
This makes sense now, thank you!